pontoon/base/views.py

```diff
@@ -158,9 +159,51 @@ def projects(request, template='projects.html'):
     return render(request, template, data)
 
 
-def translate(request, locale, slug, part=None, template='translate.html'):
+def locale_project(request, locale, slug):
+    """Locale-project overview."""
+
+    # Redirect to translate view
+    if request.GET.get('q'):
+        return translate(request, locale, slug, '')
+
+    try:
+        l = Locale.objects.get(code__iexact=locale)
+    except Locale.DoesNotExist:
+        raise Http404
+
+    # Prefetch subpages and resources for locales_parts_stats()
+    projects = (
+        Project.objects.filter(
+            disabled=False,
+            slug=slug,
+            pk__in=Resource.objects.values('project')
+        )
+        .prefetch_related(
+            'subpage_set',
+            Prefetch(
+                'resource_set',
+                queryset=Resource.objects.filter(
+                    pk__in=Entity.objects.filter(obsolete=False).values('resource')
+                ),
+                to_attr='active_resources'
+            )
+        )
+    )
+
+    if not projects.exists():
+        raise Http404
+
+    data = {
+        'locale': l,
+        'project': projects[0],
```

Comment:
Don't bother with the `filter` and indexing if you're just going to take a single project. Use `get_object_or_404` and just provide a queryset set up with the prefetching to it.